pkg/controller/networkpolicy/controller_query_test.go

```diff
@@ -0,0 +1,191 @@
+			Name:      "podB",
+			Namespace: "testNamespace",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name: "container-1",
+			}},
+			NodeName: "nodeA",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+			PodIP: "1.2.3.4",
+		},
+	},
+}
+// polices represent kubernetes policies for testing proper query results
+//
+// policy 0: select all pods and deny default ingress
+// policy 1: select all pods and deny default egress
+
+var policies = []networkingv1.NetworkPolicy{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default-deny-ingress",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+			},
+		},
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: 		"default-deny-egress",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	},
+}
+
+var namespaces = []v1.Namespace{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:                       "testNamespace",
+			UID:                        "testNamespaceUID",
+		},
+	},
+}
+
+func makeControllerAndEndpointQueryReplier(objects ...runtime.Object) (*networkPolicyController, *EndpointQueryReplier){
+	// create controller
+	_, controller := newController(objects...)
+	// create querier with stores inside controller
+	address, appliedToGroup, policy, podInformer := controller.addressGroupStore, controller.appliedToGroupStore, controller.internalNetworkPolicyStore, controller.podInformer
+	querier := NewEndpointQueryReplier(address, appliedToGroup, policy, podInformer)
+
+	return controller, querier
+}
+
+// TestInvalidSelector tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) does not
+// select any pods
+func TestInvalidSelector(t *testing.T) {
+	_, endpointQuerier := makeControllerAndEndpointQueryReplier()
+	// test appropriate response to QueryNetworkPolices
+	namespace, pod := "non-existing-namespace", "non-existing-pod"
+	response := endpointQuerier.QueryNetworkPolicies(namespace, pod)
+
+	assert.Equal(t, errors.NewNotFound(v1.Resource("pod"), pod), response.Error, "expected not found error")
+}
+
+// TestSingleAppliedPolicy tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) selects a
+// pod which has a single networkpolicy object applied to it
+func TestSingleAppliedPolicy(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	controller, endpointQuerier := makeControllerAndEndpointQueryReplier(&namespaces[0], &pods[0], &policies[0])
+
+	controller.informerFactory.Start(ctx.Done())
+	controller.crdInformerFactory.Start(ctx.Done())
+
+	go controller.NetworkPolicyController.Run(ctx.Done())
+
+	time.Sleep(2 * time.Second)
```

Comment:
add a TODO there. I think the sample benchmark that you have been looking for has a specific way of ensuring that the policies have been "realized" (i.e. the stores have been updated). So we should get rid of the sleep before we open the PR against upstream Antrea.